pkg/swap/http.go

```diff
@@ -78,11 +78,7 @@ func httpClientWithTransport(baseURL *url.URL, c *http.Client) *http.Client {
 
 	c.Transport = roundTripperFunc(func(r *http.Request) (resp *http.Response, err error) {
 		r.Header.Set("User-Agent", userAgent)
-		u, err := baseURL.Parse(r.URL.String())
-		if err != nil {
-			return nil, err
-		}
-		r.URL = u
+		r.URL = baseURL
```

Comment:
Is it ok to ignore the original r.URL and always use baseURL instead? It looks to me that url.Parse basically merges two urls.

Reply:
Now, the full URL provided in the request (including the path and query parameters) is preserved, which was not the case earlier. Also, geth-url is provided in init of the NewGethClient, so the baseURL always stays the same. This url is RPC endpoint and it doesn't change.

Reply:
Thanks, makes sense.